cli/cmd/addon_download.go

```diff
@@ -73,13 +72,9 @@ func runAddonDownload(ctx context.Context, cmd *cobra.Command, args []string) er
 		return fmt.Errorf("invalid addon ref %q: version must start with 'v'", args[0])
 	}
 
-	c, err := cloudquery_api.NewClientWithResponses(managedplugin.APIBaseURL(),
-		cloudquery_api.WithRequestEditorFn(func(ctx context.Context, req *http.Request) error {
-			req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token))
-			return nil
-		}))
+	c, err := api.NewClient(getEnvOrDefault(envAPIURL, defaultAPIURL), token.Value)
 	if err != nil {
-		return fmt.Errorf("failed to create hub client: %w", err)
+		return nil
```

Comment:
Isn't it a problem if we couldn't create the API client? What if `c` is nil?

Reply:
Oops, this is a mistake. I did this refactoring in https://github.com/cloudquery/cloudquery/pull/16230 which is OK in that PR (I return the error).

I'll rebase to remove this diff